Project/Modules/Controllers/PostsController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using blog_net_core.Project.Modules.Model.Entities;
+using blog_net_core.Project.Modules.Model.Repositories;
+using blog_net_core.Project.Modules.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace blog_net_core.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class PostsController : ControllerBase
+    {
+        private readonly IPostService _postService;
+
+        public PostsController(IPostService postService)
+        {
+            _postService = postService;
+        }
+
+        [Route("")]
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var posts = _postService.getAllPosts();
+            return Ok(posts);
+        }
+
+        [Route("{postId}")]
+        [HttpGet]
+        public IActionResult Get(int postId)
+        {
+            var post = _postService.getPostById(postId); 
+            return Ok(post);
+        }
+
+        [Route("post")]
+        [HttpPost]
+        public IActionResult Create(PostModel model)
+        {
+            var post = _postService.createPost(model);
+            return Ok(post);
+        }
+
+        [Route("update")]
+        [HttpPut]
+        public IActionResult Update(PostModel model)
+        {
+            var post = _postService.updatePost(model);
+            return Ok(post);
+        }
+        
+        [Route("delete-bd/{postId}")]
```

Comment:
remove the "delete-bd/", by "{postId:int}" because it is redundant, you can also add the type at the route level.

Reply:
corrected